### PR TITLE
Conference room model

### PR DESCRIPTION
**Pull request by jtkpiotr**
@adamzima @s1mplex @quirk0o @pbzymek @lkudyba @mhib

### CODE REVIEW

File: app/models/conference_room.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class ConferenceRoom < ApplicationRecord
+  before_validation { self.color&.downcase! }
```
Comment (s1mplex) [inline app/models/conference_room.rb]: You don't need `self` here

File: app/models/conference_room.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+  validates :capacity, presence: true
+  #validates that color format is like: #FF0ddE
```
Comment (s1mplex) [inline app/models/conference_room.rb]: A comment saying what a piece of code does usually means there's something wrong with the code ;) You could extract the regexp to a constant named `HEX_COLOR_PATTERN` or something similar

File: db/seeds.rb [hunk 3]
```diff
@@ -7,7 +7,15 @@
 #   Character.create(name: 'Luke', movie: movies.first)
 
 day_time = Time.new(2016,7,11,8,0,0,"+02:00")
-locations = ["Mordor", "Neverland", "Sherwood Forest", "Valhalla", "Voyager", "Winterfell", "Yellow Submarine"]
+locations = ["Mordor", "Neverland", "Sherwood Forest", "Valhalla", "Voyager", "Winterfell", "Yellow Submarine"].map do |n|
+  params = { title: n, capacity: rand(5.20), color: "#%06x" % (rand * 0xffffff) }
+  if (c = ConferenceRoom.find_by_title(n))
```
Comment (s1mplex) [inline db/seeds.rb]: `c = ConferenceRoom.first_or_create(title: n)` will get rid of the conditional

File: db/seeds.rb [hunk 4]
```diff
@@ -7,7 +7,15 @@
 #   Character.create(name: 'Luke', movie: movies.first)
 
 day_time = Time.new(2016,7,11,8,0,0,"+02:00")
-locations = ["Mordor", "Neverland", "Sherwood Forest", "Valhalla", "Voyager", "Winterfell", "Yellow Submarine"]
+locations = ["Mordor", "Neverland", "Sherwood Forest", "Valhalla", "Voyager", "Winterfell", "Yellow Submarine"].map do |n|
```
Comment (s1mplex) [inline db/seeds.rb]: `n` is rather a bad variable name

Comment (s1mplex) [inline db/seeds.rb]: `locations` should now be named `conference_rooms` (or you can 2 variables)

File: db/seeds.rb [hunk 5]
```diff
@@ -7,7 +7,15 @@
...
+  params = { title: n, capacity: rand(5.20), color: "#%06x" % (rand * 0xffffff) }
+  if (c = ConferenceRoom.find_by_title(n))
+    c.update_attributes!(**params)
+    c
+  else
+    ConferenceRoom.create!(**params)
```
Comment (s1mplex) [inline db/seeds.rb]: You don't need to use `**` operator neither in `create!` nor `update_attributes!`

File: db/seeds.rb [hunk 6]
```diff
@@ -16,12 +24,17 @@
...
-      event.location = locations.sample
-      event.user = Faker::Name.name
+    params = {
+        start_time: start_time,
+        end_time: end_time,
+        description: Faker::Company.catch_phrase,
+        conference_room: locations.sample,
+        user: Faker::Name.name
+    }
+    if (e = Event.find_by_id(event_id))
```
Comment (s1mplex) [inline db/seeds.rb]: Use `first_or_create` ;)

File: spec/models/conference_room_spec.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+RSpec.describe ConferenceRoom, type: :model do
+  describe "validation" do
+
```
Comment (s1mplex) [inline spec/models/conference_room_spec.rb]: This file has poor formatting:
1. multiple newlines
2. lack of newlines  (eg. lines 29-30)
3. different formatting conventions in `describe` blocks

File: db/seeds.rb [hunk 8]
```diff
@@ -7,7 +7,15 @@
 #   Character.create(name: 'Luke', movie: movies.first)
 
 day_time = Time.new(2016,7,11,8,0,0,"+02:00")
-locations = ["Mordor", "Neverland", "Sherwood Forest", "Valhalla", "Voyager", "Winterfell", "Yellow Submarine"]
+locations = ["Mordor", "Neverland", "Sherwood Forest", "Valhalla", "Voyager", "Winterfell", "Yellow Submarine"].map do |n|
+  params = { title: n, capacity: rand(5.20), color: "#%06x" % (rand * 0xffffff) }
+  if (c = ConferenceRoom.find_by_title(n))
+    c.update_attributes!(**params)
```
Comment (pbzymek) [inline db/seeds.rb]: this is old style (and alias), you should use update/update!
`c.update!(params)`

File: spec/factories/conference_rooms.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+FactoryGirl.define do
+  factory :conference_room do
+    capacity 1
+    title "MyString"
```
Comment (pbzymek) [inline spec/factories/conference_rooms.rb]: use sequence here due to uniqueness validation on those fields:
`sequence :title {|n| "Conf Room ##{n}" `
`sequence :color {|n| "#fff00#{n}" `

File: spec/factories/event_factory.rb [hunk 10]
```diff
@@ -2,8 +2,8 @@
   factory :event do
     description "123"
     user "user"
-    location "location"
     start_time Time.now
     end_time Time.now + 10.minutes
+    conference_room
```
Comment (pbzymek) [inline spec/factories/event_factory.rb]: this should be an association:
`association : conference_room, factory: :conference_room `

File: spec/models/conference_room_spec.rb [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+      expect(invalid_colors.none? { |n| subject.tap { |s| s.color = n }.valid? }).to be true
+    end
+
+  end
+
+  describe "before_validation" do
+    let(:color) { "#fFB021" }
+    let(:conference_room) { build :conference_room, color: color }
+    it "must downcases color" do
+      conference_room.valid? # Triggers before_validation
```
Comment (pbzymek) [inline spec/models/conference_room_spec.rb]: I usually put the "set up" code in `before` section

**Comment by lkudyba** [comment]
👍

**Pull request merged**